File: root/trunk/source/modules/strategy/src/test/sample_data/inputs/suggest_trades.rb
include_class "org.marketcetera.strategy.ruby.Strategy"
include_class "java.math.BigDecimal"
include_class "org.marketcetera.trade.Factory"
include_class "org.marketcetera.trade.OrderType"
include_class "org.marketcetera.trade.Side"
include_class "org.marketcetera.trade.TimeInForce"
include_class "org.marketcetera.core.MSymbol"

class SuggestTrades < Strategy
    def on_start
        orderShouldBeNull = get_parameter("orderShouldBeNull")
        if(orderShouldBeNull != nil)
            suggest_trade(nil, BigDecimal.new("100.00"), "suggestion")
        else
            suggestion = Factory.getInstance().createOrderSingleSuggestion()
            suggestion.setAccount(get_parameter("account"))
            orderType = get_parameter("orderType")
            if(orderType != nil)
                suggestion.setOrderType(OrderType.valueOf(orderType))
            end
            price = get_parameter("price")
            if(price != nil)
                suggestion.setPrice(BigDecimal.new(price))
            end
            quantity = get_parameter("quantity")
            if(quantity != nil)
                suggestion.setQuantity(BigDecimal.new(quantity))
            end
            side = get_parameter("side")
            if(side != nil)
                suggestion.setSide(Side.valueOf(side))
            end
            symbol = get_parameter("symbol")
            if(symbol != nil)
                suggestion.setSymbol(MSymbol.new(symbol))
            end
            timeInForce = get_parameter("timeInForce")
            if(timeInForce != nil)
                suggestion.setTimeInForce(TimeInForce.valueOf(timeInForce))
            end
            scoreString = get_parameter("score")
            if(scoreString != nil)
                score = BigDecimal.new(scoreString)
            else
                score = nil
            end
            suggest_trade(suggestion, score, get_parameter("identifier"))
        end
    end
end

File: root/branches/will/eclipse35/source/modules/strategy/src/test/sample_data/inputs/SuggestTrades.java
import java.math.BigDecimal;

import org.marketcetera.trade.MSymbol;
import org.marketcetera.strategy.java.Strategy;
import org.marketcetera.trade.Factory;
import org.marketcetera.trade.OrderSingle;
import org.marketcetera.trade.OrderType;
import org.marketcetera.trade.Side;
import org.marketcetera.trade.TimeInForce;

/**
 * Test strategy that suggests trades.
 *
 * @author <a href="mailto:colin@marketcetera.com">Colin DuPlantis</a>
 * @version $Id$
 * @since 1.0.0
 */
public class SuggestTrades
    extends Strategy
{
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStart()
     */
    @Override
    public void onStart()
    {
        String orderShouldBeNull = getParameter("orderShouldBeNull");
        if(orderShouldBeNull != null) {
            suggestTrade(null,
                         new BigDecimal("100.00"),
                         "suggestion");
        } else {
            OrderSingle suggestedOrder = Factory.getInstance().createOrderSingle();
            suggestedOrder.setAccount(getParameter("account"));
            String orderType = getParameter("orderType");
            if(orderType != null) {
                suggestedOrder.setOrderType(OrderType.valueOf(orderType));
            }
            String price = getParameter("price");
            if(price != null) {
                suggestedOrder.setPrice(new BigDecimal(price));
            }
            String quantity = getParameter("quantity");
            if(quantity != null) {
                suggestedOrder.setQuantity(new BigDecimal(quantity));
            }
            String side = getParameter("side");
            if(side != null) {
                suggestedOrder.setSide(Side.valueOf(side));
            }
            String symbol = getParameter("symbol");
            if(symbol != null) {
                suggestedOrder.setSymbol(new MSymbol(symbol));
            }
            String timeInForce = getParameter("timeInForce");
            if(timeInForce != null) {
                suggestedOrder.setTimeInForce(TimeInForce.valueOf(timeInForce));
            }
            String scoreString = getParameter("score");
            BigDecimal score;
            if(scoreString != null) {
                score = new BigDecimal(scoreString);
            } else {
                score = null;
            }
            if(getParameter("doOrder") != null) {
                if(getParameter("sendOrderShouldBeNull") != null) {
                    sendOrder(null);
                } else {
                    sendOrder(suggestedOrder);
                }
            } else {
                suggestTrade(suggestedOrder,
                             score,
                             getParameter("identifier"));
            }
        }
    }
}